app/parse.py

```diff
@@ -14,5 +19,47 @@ class Course:
     course_type: CourseType
 
 
+def parse_single_course(course_soup: BeautifulSoup,
+                        ) -> list[Course]:
+    courses = []
+
+    button_texts = [
+        button.text
+        for button in course_soup.select(".ButtonBody_buttonText__FMZEg")
+    ]
+    if "Full time" in button_texts:
+        append_course(course_soup, courses, CourseType.FULL_TIME)
+    if "Flex" in button_texts:
+        append_course(course_soup, courses, CourseType.PART_TIME)
+    return courses
```

Comment:
This function looks, like it processes several courses (because it returns a list of courses), but as it's named -> it should process exactly one course. 


Also, append_course is not really obvious from the first time of view. 

Try to rewrite this function to really process only one course - then you will be able to remove append_course function. Just create a course object right in this function and return it. 
Course type can be defined to some variable and then just create course not to duplicate course creation code.